pkg/applicationserver/applicationserver.go

```diff
@@ -301,6 +301,7 @@ func (as *ApplicationServer) processUp(ctx context.Context, up *ttnpb.Applicatio
 }
 
 func (as *ApplicationServer) publishUp(ctx context.Context, up *ttnpb.ApplicationUp) error {
+	ctx = as.FromRequestContext(ctx)
```

Comment:
I don't find this a logical change. By convention, the caller provides the context.

Reply:
The caller provides the context indeed, but having to do `as.FromRequestContext` each time you call `publishUp` is error prone, and before this PR a call site was missed. Nonetheless, I'll change this back to call sites.

Reply:
Not each time: only when the caller is actually aware of the fact that the context decouples.